compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -97,6 +97,16 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
 
                         ControlFlow::CONTINUE
                     }
+                    Node::Cast(_, _, ty) => {
+                        let ty = ty.subst(tcx, ct.substs);
+                        if ty.has_infer_types_or_consts() {
+                            failure_kind = FailureKind::MentionsInfer;
+                        } else if ty.has_param_types_or_consts() {
+                            failure_kind = cmp::min(failure_kind, FailureKind::MentionsParam);
+                        }
+
+                        ControlFlow::CONTINUE
+                    }
```

Comment:
is this necessary?

Reply:
casts can be `as _`, so I guess, yes? Not sure if they can still be that at the MIR level though

Reply:
lol `as _` doesnt even work because the mir lowering does something like
```
_1 = N as u128
_0 = _1
```
[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c350c1cba6e28c9dc60212e30203e121)
and we dont support field accesses in abstract consts yet ^^

Reply:
time for more tests 🙃 I think we can ignore the ascribeusertype statement in abstract const generation

Reply:
and a whole lot of tests later `as _` is working :sweat_smile: